yt_dlp/extractor/zingmp3.py

```diff
@@ -51,45 +83,88 @@ def _extract_item(self, item, fatal):
+                'id': item_id})
+            info_lyric = self.download_json(api, item_id)
+            lyric = traverse_obj(info_lyric, ('data', 'file'))
+        subtitles = {
+            'origin': [{
+                'url': lyric,
+            }],
+        } if lyric else None
 
         album = item.get('album') or {}
 
         return {
             'id': item_id,
             'title': title,
             'formats': formats,
-            'thumbnail': item.get('thumbnail'),
+            'thumbnail': item.get('thumbnail') or item.get('thumbnailM'),
             'subtitles': subtitles,
             'duration': int_or_none(item.get('duration')),
             'track': title,
-            'artist': item.get('artists_names'),
+            'artist': item.get('artistsNames') or item.get('artists_names'),
             'album': album.get('name') or album.get('title'),
-            'album_artist': album.get('artists_names'),
+            'album_artist': album.get('artistsNames') or album.get('artists_names'),
         }
 
     def _real_extract(self, url):
-        page_id = self._match_id(url)
-        webpage = self._download_webpage(
-            url.replace('://zingmp3.vn/', '://mp3.zing.vn/'),
-            page_id, query={'play_song': 1})
-        data_path = self._search_regex(
-            r'data-xml="([^"]+)', webpage, 'data path')
-        return self._process_data(self._download_json(
-            'https://mp3.zing.vn/xhr' + data_path, page_id)['data'])
+        song_id, type_url = self._match_valid_url(url).group('id', 'type')
+
+        slug_api = SLUG_API.get(type_url)
+
+        # Create api from slug_api with hash signature
+        api = self.get_api_with_signature(name_api=slug_api, param={
+            'ctime': TIMESTAMP,
+            'id': song_id})
+        info = self.download_json(api, song_id)
+        return self._process_data(info.get('data'), song_id, type_url)
+
+    def download_json(self, api, video_id):
+        # For the first time making the request, we need to make one more temp request to update the cookie
+        # and the next time we don't need to do it.
+        # But if download with cookieFile, so we don't need to make one more temp request
+        if self._downloader.params.get('cookiefile'):
+            self._IS_UPDATE_COOKIES = True
+
+        if not self._IS_UPDATE_COOKIES:
+            self._download_json(api, video_id=video_id, note='Update cookies for request')
+            self._IS_UPDATE_COOKIES = True
+
+        return self._download_json(api, video_id=video_id)
+
+    def get_api_with_signature(self, name_api, param):
+        if not name_api:
+            return
+
+        def get_hash256(string):
+            return hashlib.sha256(string.encode('utf-8')).hexdigest()
+
+        def get_hmac512(string):
+            return hmac.new(SECRET_KEY, string.encode('utf-8'), hashlib.sha512).hexdigest()
+
+        text = ''
+        for k, v in param.items():
+            text += f'{k}={v}'
```

Comment:
json.dumps(param).replace('\"', '').replace(',','') will do the same thing ig but this is okay too